### PR TITLE
Add contributing style guide

### PR DESCRIPTION
**Pull request by soapy1**

### CODE REVIEW

**Review by briancain — CHANGES_REQUESTED** [review]

File: .github/CONTRIBUTING.md [hunk 1]
```diff
@@ -30,7 +32,39 @@ No pull request template is provided on GitHub. The expected changes are often a
...
+It's nice to have a way to control what version of ruby is installed, so you may want to install [rvm](https://rvm.io/rvm/install), [chruby](https://github.com/postmodern/chruby#install) or [rbenv](https://github.com/rbenv/rbenv#installation). For Windows [ruby installer](https://rubyinstaller.org/) is recommended.
+
+#### Setup Vagrant
+Clone Vagrant's repository from GitHub into the directory where you keep code on your machine:
+
+```
+  $ git clone https://github.com/hashicorp/vagrant.git
+```
+
+Next, cd into that path. All commands will be run from this path:
```
Comment (chrisroberts) [inline .github/CONTRIBUTING.md:53]: Next, move into the newly created `./vagrant directory:

```
$ cd ./vagrant
```

All commands will be run from this path. Now, run the `bundle` command to install the Ruby dependencies:

```
$ bundle install
```

File: .github/CONTRIBUTING.md [hunk 2]
```diff
@@ -44,6 +78,49 @@ Once you're confident that your upcoming changes will be accepted:
 * If possible, submit tests along with your topic branch. It will help a lot to get your your patch / new feature accepted, and should prevent unwanted breaking changes to silently happen in future developments.
```
Comment (chrisroberts) [inline .github/CONTRIBUTING.md:78]: We should update the wording here as tests being _required_ unless they are not applicable for some reason (like docs changes, dependency constraint updates, etc).

File: .github/CONTRIBUTING.md [hunk 3]
```diff
@@ -44,6 +78,49 @@ Once you're confident that your upcoming changes will be accepted:
...
 
+### Running tests
+
+Vagrant uses rspec to run tests. Once your Vagrant bundle is installed from Git repository, you can run the test suite with:
+
+    bundle exec rake
+
+This will run the unit test suite, which should come back all green!
+
+If you are developing Vagrant on a machine that already has a Vagrant package installation present, both will attempt to use the same folder for their configuration (location of this folder depends on system). This can cause errors when Vagrant attempts to load plugins. In this case, override the `VAGRANT_HOME` environment variable for your development version of Vagrant before running any commands, to be some new folder within the project or elsewhere on your machine. For example, in Bash:
```
Comment (chrisroberts) [inline .github/CONTRIBUTING.md:89]: > before running any commands, to be some new

That comma can be removed.

**Review by chrisroberts — CHANGES_REQUESTED** [review]
This looks great!

File: .github/CONTRIBUTING.md [hunk 4]
```diff
@@ -30,7 +32,40 @@ No pull request template is provided on GitHub. The expected changes are often a
...
+It's nice to have a way to control what version of ruby is installed, so you may want to install [rvm](https://rvm.io/rvm/install), [chruby](https://github.com/postmodern/chruby#install) or [rbenv](https://github.com/rbenv/rbenv#installation). For Windows [ruby installer](https://rubyinstaller.org/) is recommended.
+
+#### Setup Vagrant
+Clone Vagrant's repository from GitHub into the directory where you keep code on your machine:
+
+```
+  $ git clone https://github.com/hashicorp/vagrant.git
+```
+
+Next, move into the newly create `./vagrant` directory. All commands will be run from this path:
```
Comment (chrisroberts) [inline .github/CONTRIBUTING.md:54]: ~All commands will be run from this path:~

`create` -> `created`

**Review by chrisroberts — DISMISSED** [review]
Two small quick changes but feel free to merge after.

**Pull request merged**